ctx/ctx.go

```diff
@@ -133,3 +136,38 @@ func GetDefinition(r *http.Request) *apidef.APIDefinition {
 	}
 	return nil
 }
+
+// GetOASDefinition returns a copy of the OAS definition of the called API.
+func GetOASDefinition(r *http.Request) *oas.OAS {
+	v := r.Context().Value(OASDefinition)
+	if v == nil {
+		return nil
+	}
+
+	val, ok := v.(*oas.OAS)
+	if !ok {
+		return nil
+	}
+
+	ret, err := cloneOAS(val)
+	if err != nil {
+		logger.Get().Error("Cloning OAS object in the request context")
+	}
+
+	return ret
+}
+
+func cloneOAS(val *oas.OAS) (*oas.OAS, error) {
```

Comment:
So, let's move this to apidef/oas, and maybe make it a public Clone(*OAS) (*OAS, error) - it should be a part of the apidef/oas apis, all it uses is the OAS type. Including the ctx package should not be a problem from apidef/oas